Using a secret from `setSecret` with `withRegistryAuth` fails

@vikram-dagger tried to use a new version of secrets in the registry authentication and it failed:

Here is a repro
```go
package main

import (
	"context"
	"fmt"
	"os"

	"dagger.io/dagger"
)

func main() {
	ctx := context.Background()
	c, err := dagger.Connect(ctx, dagger.WithLogOutput(os.Stderr))
	if err != nil {
		panic(err)
	}

	sec := c.SetSecret("my-secret-id", "yolo")
	stdout, err := c.Container().WithRegistryAuth("localhost:8888", "YOLO", sec).Publish(ctx, "localhost:8888/myorg/myapp")
	if err != nil {
		panic(err)
	}
	fmt.Println(stdout)
}
```

Result:
```console
panic: input:1: container.withRegistryAuth plaintext: empty secret?

Please visit https://dagger.io/help#go for troubleshooting guidance.

goroutine 1 [running]:
main.main()
	/home/dolanor/src/daggerr/vikram-setsecret-empty/main.go:21 +0x1b1
exit status 2
```

--- a/core/integration/container_test.go
+++ b/core/integration/container_test.go
@@ -2768,21 +2768,36 @@ func TestContainerWithRegistryAuth(t *testing.T) {
 	_, err = container.Publish(ctx, testRef)
 	require.Error(t, err)
 
-	pushedRef, err := container.
-		WithRegistryAuth(
-			privateRegistryHost,
-			"john",
-			//nolint:staticcheck // SA1019 We want to test this API while we support it.
-			c.Container().
-				WithNewFile("secret.txt", dagger.ContainerWithNewFileOpts{Contents: "xFlejaPdjrt25Dvr"}).
-				File("secret.txt").
-				Secret(),
-		).
-		Publish(ctx, testRef)
+	t.Run("legacy secret API", func(t *testing.T) {
+		pushedRef, err := container.
+			WithRegistryAuth(
+				privateRegistryHost,
+				"john",
+				//nolint:staticcheck // SA1019 We want to test this API while we support it.
+				c.Container().
+					WithNewFile("secret.txt", dagger.ContainerWithNewFileOpts{Contents: "xFlejaPdjrt25Dvr"}).
+					File("secret.txt").
+					Secret(),
+			).
+			Publish(ctx, testRef)
 
-	require.NoError(t, err)
-	require.NotEqual(t, testRef, pushedRef)
-	require.Contains(t, pushedRef, "@sha256:")
+		require.NoError(t, err)
+		require.NotEqual(t, testRef, pushedRef)
+		require.Contains(t, pushedRef, "@sha256:")
+	})
+	t.Run("new secret API", func(t *testing.T) {
+		pushedRef, err := container.
+			WithRegistryAuth(
+				privateRegistryHost,
+				"john",
+				c.SetSecret("this-secret", "xFlejaPdjrt25Dvr"),
+			).
+			Publish(ctx, testRef)
+
+		require.NoError(t, err)
+		require.NotEqual(t, testRef, pushedRef)
+		require.Contains(t, pushedRef, "@sha256:")
+	})
 }
 
 func TestContainerImageRef(t *testing.T) {

--- a/core/schema/container.go
+++ b/core/schema/container.go
@@ -580,12 +580,12 @@ type containerWithRegistryAuthArgs struct {
 }
 
 func (s *containerSchema) withRegistryAuth(ctx *router.Context, parents *core.Container, args containerWithRegistryAuthArgs) (*core.Container, error) {
-	secret, err := core.NewSecret(args.Secret).Plaintext(ctx, s.gw)
+	secretBytes, err := s.secrets.GetSecret(ctx, args.Secret.String())
 	if err != nil {
 		return nil, err
 	}
 
-	if err := s.auth.AddCredential(args.Address, args.Username, string(secret)); err != nil {
+	if err := s.auth.AddCredential(args.Address, args.Username, string(secretBytes)); err != nil {
 		return nil, err
 	}
 

